File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/rds/RDSConfigProvider.java
package com.sequenceiq.cloudbreak.blueprint.rds;


import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.blueprint.BlueprintComponentConfigProvider;
import com.sequenceiq.cloudbreak.blueprint.BlueprintConfigurationEntry;
import com.sequenceiq.cloudbreak.blueprint.BlueprintPreparationObject;
import com.sequenceiq.cloudbreak.blueprint.BlueprintProcessor;
import com.sequenceiq.cloudbreak.domain.RDSConfig;

@Component
public class RDSConfigProvider implements BlueprintComponentConfigProvider {

    @Inject
    private BlueprintProcessor blueprintProcessor;

    @Override
    public String configure(BlueprintPreparationObject source, String blueprintText) {
        List<BlueprintConfigurationEntry> bpConfigs = new ArrayList<>();
        for (RDSConfig rds : source.getRdsConfigs()) {
            switch (rds.getType()) {
                case HIVE:
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionURL",
                            rds.getConnectionURL()));
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionDriverName",
                            rds.getDatabaseType().getDbDriver()));
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionUserName",
                            rds.getConnectionUserName()));
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionPassword",
                            rds.getConnectionPassword()));
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-env", "hive_database",
                            rds.getDatabaseType().getAmbariDbOption()));
                    bpConfigs.add(new BlueprintConfigurationEntry("hive-env", "hive_database_type",
                            rds.getDatabaseType().getDbName()));
                    break;
                case RANGER:
                    break;
                case DRUID:
                    bpConfigs.add(new BlueprintConfigurationEntry("druid-common", "druid.metadata.storage.type",
                            parseDatabaseTypeFromJdbcUrl(rds.getConnectionURL())));
                    bpConfigs.add(new BlueprintConfigurationEntry("druid-common", "druid.metadata.storage.connector.connectURI",
                            rds.getConnectionURL()));
                    bpConfigs.add(new BlueprintConfigurationEntry("druid-common", "druid.metadata.storage.connector.user",
                            rds.getConnectionUserName()));
                    bpConfigs.add(new BlueprintConfigurationEntry("druid-common", "druid.metadata.storage.connector.password",
                            rds.getConnectionPassword()));
                    break;
                default:
                    break;
            }
        }
        return blueprintProcessor.addConfigEntries(blueprintText, bpConfigs, true);
    }

    @Override
    public boolean additionalCriteria(BlueprintPreparationObject source, String blueprintText) {
        return true;
    }

    private String parseDatabaseTypeFromJdbcUrl(String jdbcUrl) {
        return jdbcUrl.split(":")[1];
    }

}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/blueprint/RDSConfigProvider.kt
package com.sequenceiq.cloudbreak.service.cluster.flow.blueprint


import java.util.ArrayList

import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.domain.RDSConfig

@Component
class RDSConfigProvider {

    fun getConfigs(rdsConfig: RDSConfig): List<BlueprintConfigurationEntry> {
        val bpConfigs = ArrayList<BlueprintConfigurationEntry>()
        bpConfigs.add(BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionURL", rdsConfig.connectionURL))
        bpConfigs.add(BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionDriverName", rdsConfig.databaseType.dbDriver))
        bpConfigs.add(BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionUserName", rdsConfig.connectionUserName))
        bpConfigs.add(BlueprintConfigurationEntry("hive-site", "javax.jdo.option.ConnectionPassword", rdsConfig.connectionPassword))
        bpConfigs.add(BlueprintConfigurationEntry("hive-env", "hive_database", rdsConfig.databaseType.ambariDbOption))
        bpConfigs.add(BlueprintConfigurationEntry("hive-env", "hive_database_type", rdsConfig.databaseType.dbName))
        return bpConfigs
    }

}

